File: T2MA22.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author USUARIO
 */
import java.io.File;
import java.util.Scanner;
import java.util.Locale;

public class T2MA22 {
    public static void main(String[] args) {
        // Ruta absoluta del archivo
        String ruta = "C:\\Users\\USUARIO\\Desktop\\TRABAJOS UTA KARLA GUANOTOA\\productos.txt";
        
        double totalVenta = 0;
        double maxSubtotal = -1;
        String prodMayor = "";

        System.out.println("--- DETALLE DE PRODUCTOS ---");
        System.out.println("-------------------------------------------");

        try {
            File archivo = new File(ruta);
            Scanner lector = new Scanner(archivo).useLocale(Locale.US);

            while (lector.hasNext()) {
                String nombreProd = lector.next();
                
                if (lector.hasNextDouble()) {
                    double precio = lector.nextDouble();
                    int cantidad = lector.nextInt();

                    double subtotal = precio * cantidad;
                    
                    System.out.printf("Producto: %-12s | Subtotal: $%.2f%n", nombreProd, subtotal);

                    totalVenta += subtotal;

                    if (subtotal > maxSubtotal) {
                        maxSubtotal = subtotal;
                        prodMayor = nombreProd;
                    }
                }
            }
            lector.close();

            System.out.println("-------------------------------------------");
            if (totalVenta > 0) {
                System.out.printf("TOTAL GENERAL DE LA VENTA: $%.2f%n", totalVenta);
                System.out.printf("PRODUCTO CON MAYOR GASTO:  %s ($%.2f)%n", prodMayor, maxSubtotal);
            }

        } catch (Exception e) {
            System.out.println("Error al leer el archivo. Revisa la ruta o el formato.");
        }
    }
}

File: T2MA22.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <iomanip>

using namespace std;

int main() {
    ifstream archivo("productos.txt");

    string nombreProd, prodMayor;
    double precio, subtotal, totalVenta = 0, maxSubtotal = -1;
    int cantidad;

    if (!archivo.is_open()) {
        cout << "--- ERROR: No se puede abrir 'productos.txt' ---" << endl;
        cout << "Asegurate de que el archivo este en la misma carpeta que este codigo." << endl;
        return 1;
    }

    cout << "--- DETALLE DE PRODUCTOS ---" << endl;
    cout << fixed << setprecision(2);

    while (archivo >> nombreProd >> precio >> cantidad) {
        subtotal = precio * cantidad;
        cout << "Producto: " << left << setw(12) << nombreProd
             << " | Subtotal: $" << subtotal << endl;

        totalVenta += subtotal;

        if (subtotal > maxSubtotal) {
            maxSubtotal = subtotal;
            prodMayor = nombreProd;
        }
    }

    cout << "-------------------------------------------" << endl;

    if (totalVenta > 0) {
        cout << "TOTAL GENERAL DE LA VENTA: $" << totalVenta << endl;
        cout << "PRODUCTO CON MAYOR GASTO:  " << prodMayor << " ($" << maxSubtotal << ")" << endl;
    } else {
        cout << "AVISO: El archivo se abrio pero no se procesaron datos." << endl;
    }

    archivo.close();
    return 0;
}
